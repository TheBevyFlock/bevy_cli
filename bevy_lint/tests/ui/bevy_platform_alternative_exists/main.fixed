#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::bevy_platform_alternative_exists)]

#[allow(dead_code)]
struct Player {
    attack_cd: bevy::platform::time::Instant,
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant
}

fn main() {
    let mut hash_map = bevy::platform::collections::HashMap::new();
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::collections::HashMap` can be replaced with the `no_std` compatible
    // type bevy::platform::collections::HashMap
    hash_map.insert("foo", "bar");

    let _time = bevy::platform::time::Instant::now();
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant

    // This should be fine even tho it will result in a `std::time::Instant` after full path
    // resolution.
    let _bevy_time = bevy::platform::time::Instant::now();
}
