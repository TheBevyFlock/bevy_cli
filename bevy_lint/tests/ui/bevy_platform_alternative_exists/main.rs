#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::bevy_platform_alternative_exists)]

use std::time;

#[allow(dead_code)]
struct Player {
    attack_cd: time::Instant,
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant
}

fn main() {
    let mut hash_map = std::collections::HashMap::new();
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::collections::HashMap` can be replaced with the `no_std` compatible
    // type bevy::platform::collections::HashMap
    hash_map.insert("foo", "bar");

    let _time = std::time::Instant::now();
    //~^ ERROR: Used type from the `std` that has an existing alternative from `bevy_platform`
    //~| HELP: the type `std::time::Instant` can be replaced with the `no_std` compatible
    // type bevy::platform::time::Instant

    // This should be fine even tho it will result in a `std::time::Instant` after full path
    // resolution.
    let _bevy_time = bevy::platform::time::Instant::now();
}
