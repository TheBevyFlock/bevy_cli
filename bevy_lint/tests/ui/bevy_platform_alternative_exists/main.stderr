error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/main.rs:16:24
   |
16 |     let mut hash_map = std::collections::HashMap::new();
   |                        ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/bevy_platform_alternative_exists/main.rs:3:9
   |
 3 | #![deny(bevy::bevy_platform_alternative_exists)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: the type `std::collections::HashMap` can be replaced with the `no_std` compatible type bevy::platform::collections::HashMap
   |
16 -     let mut hash_map = std::collections::HashMap::new();
16 +     let mut hash_map = bevy::platform::collections::HashMap::new();
   |

error: Used type from the `std` that has an existing alternative from `bevy_platform`
  --> tests/ui/bevy_platform_alternative_exists/main.rs:22:17
   |
22 |     let _time = std::time::Instant::now();
   |                 ^^^^^^^^^^^^^^^^^^
   |
help: the type `std::time::Instant` can be replaced with the `no_std` compatible type bevy::platform::time::Instant
   |
22 -     let _time = std::time::Instant::now();
22 +     let _time = bevy::platform::time::Instant::now();
   |

error: aborting due to 2 previous errors

